src/utils/PolygonProximityLinker.cpp

```diff
@@ -189,6 +189,22 @@ void PolygonProximityLinker::findProximatePoints(const ListPolyIt a_point_it, Li
         return;
     }
 
+    if (dist2 > proximity_distance_2 * 0.9)
```

Comment:
This doesn't filter out any short line segments. It filters out line segments that are sort-of-close but not very close. I think this is then not a good change to make because it negatively affects accuracy. If they are sort-of-close but not very close (<10% overlap) then the flow rate won't change much anyway, so the visual quality will not be affected much.

You might've seen a user here that judged print quality based on how layer view looked, or based on a single experiment with one model.

The original idea is good to me, but this implementation is not implementing that idea.